tfhe/src/core_crypto/fft_impl/fft128/crypto/ggsw.rs

```diff
@@ -1,3 +1,5 @@
+#![allow(clippy::redundant_locals)]
```

Comment:
why is this needed ?

Reply:
Otherwise, the line `let output_fft_buffer_re0 = &mut *output_fft_buffer_re0;` and the 3 next ones are generating a warning whihc I believe is a false positive

Reply:
removing the &mut is not enough ? looks like deref returns a mutable ref to a slice

Reply:
basically I'm all against relaxing lints at file level, hence why I'm aksing this :)

it's an easy way to over disable stuff

Reply:
if this is really a false positive then I would prefer a function level disable or even a block disable as these rust attributes can be put in a lot of places

Reply:
> removing the &mut is not enough ? looks like deref returns a mutable ref to a slice
No, apparently it returns a plain `[f64]`


> if this is really a false positive then I would prefer a function level disable or even a block disable as these rust attributes can be put in a lot of places

Sure, I had the same idea after seeing your comment :)

Reply:
could the lines be removed maybe ? maybe the compiler can deref things on its own

Reply:
This does not work :(
I think that's the first thing I tried

Reply:
Removing the variables and adding `&mut` where they were used works (the `*` is not even needed)

Reply:
perfect